app/helpers/application_helper/button/host_toggle_maintenance.rb

```diff
@@ -0,0 +1,5 @@
+class ApplicationHelper::Button::HostToggleMaintenance < ApplicationHelper::Button::Basic
+  def disabled?
+    !(@record.is_available?(:set_node_maintenance) && @record.is_available?(:unset_node_maintenance))
+  end
```

Comment:
Also, the button tooltip needs to say something else when the button is disabled (i.e. the host does not
support maintenance mode).

So you need to add something like this to this class:

```ruby
def calculate_properties
  super
  self[:title] = N_("Maintenance mode is not supported") if disabled? # or some other nice message
end
```